### ISSUE TITLE
Dependency ordering may be OS specific

### ISSUE THREAD

**Issue by cescoffier** [fabric8io/vertx-maven-plugin#110]
When iterating over dependencies, the Maven order must be preserved in order to be OS-agnostic.

This impacts:

* the final jar when a package is available in 2 dependencies
* the SPI combination

**Comment by kameshsampath** [comment]
just putting the sample dependency content based on our discussions,

`jar1 with io.vertx.foo/foo.txt=hello`
`jar2 with io.vertx.foo/foo.txt=bonjour`

with maven pom.xml like 

```
...

<dependency>
  <artifactId>com.example</artifactId>
   <artifactId>foo1</artifactId>
  <version>1.0</version>
</dependency>

<dependency>
  <artifactId>com.test</artifactId>
   <artifactId>foo2</artifactId>
  <version>1.0</version>
</dependency>
...

```

So when SPI combination is done, the content of foo1 content should be followed by foo2 content.

**Comment by kameshsampath** [comment]
closed via #111
